### PR TITLE
Feature/issue 391 add should not receive expectation

### PR DESCRIPTION
**Pull request by robertbasic**
Took a stab at #391

### CODE REVIEW

File: library/Mockery/Mock.php [hunk 1]
```diff
@@ -203,6 +203,19 @@ public function shouldReceive()
     }
 
     /**
+     * Shortcut method for setting an expectation that a method should not be called.
+     *
+     * @param mixed
+     * @return \Mockery\Expectation
+     */
+    public function shouldNotReceive()
```
Comment (aik099) [inline library/Mockery/Mock.php]: I guess this needs to be added to `MockInterface` as well.

File: docs/reference/expectations.rst [hunk 2]
```diff
@@ -43,6 +43,12 @@ usage (e.g. during refactoring). See examples in a later section.
 
 .. code-block:: php
 
+    shouldNotReceive(method_name)
+
+Declares that the mock should not expect a call to the given method name.
```
Comment (davedevelopment) [inline docs/reference/expectations.rst]: I think a little comment here to say that it is effectively shorthand for `shouldReceive()->never()` would be good.

Comment (padraic) [inline docs/reference/expectations.rst]: :+1:

**Comment by robertbasic** [comment]
@aik099 @padraic @davedevelopment PR updated!

File: tests/Mockery/ExpectationTest.php [hunk 3]
```diff
@@ -402,6 +402,31 @@ public function testNeverCalled()
         $this->mock->shouldReceive('foo')->never();
     }
 
+    public function testShouldNotReceive()
+    {
+        $this->mock->shouldNotReceive('foo');
```
Comment (aik099) [inline tests/Mockery/ExpectationTest.php]: Should you be calling `$this->container->mockery_verify();` here as well and verify, that if more than 0 calls are made we're getting an exception?

**Comment by davedevelopment** [comment]
Good work @robertbasic

**Pull request merged**